Sources/SwiftParser/Availability.swift

```diff
@@ -48,11 +48,16 @@ extension Parser {
           )
         )
       } while keepGoing != nil && self.hasProgressed(&availabilityArgumentProgress)
+        && !self.atAvailabilitySpecListTerminator()
```

Comment:
I would prefer to keep the `hasProgressed` as the last element in the condition.

Same in the other conditions.